spec/inputs/tcp_spec.rb

```diff
@@ -269,6 +269,39 @@ def get_port
     end
   end
 
+  it "should flush codec after client disconnects" do
+    # verifies fix for https://github.com/logstash-plugins/logstash-input-tcp/issues/90
+    conf = <<-CONFIG
+      input {
+        tcp {
+          port => #{port}
+          codec => multiline {
+              pattern => "^\s"
+              what => "previous"
+          }
+        }
+      }
+    CONFIG
+
+    data = "a\n 1\n 2\nb\n 1"
+    event_count = 2
+
+    events = input(conf) do |pipeline, queue|
+      socket = Stud::try(5.times) { TCPSocket.new("127.0.0.1", port) }
+      socket.puts(data)
+      socket.close
+
+      event_count.times.collect {
+        # use non-blocking pop with sleep so that a missing queue item doesn't cause the test to hang
+        # if a "queue empty" error occurs here, the codec was not properly flushed
+        sleep(2)
+        queue.pop(true)
+      }
+    end
+
+    insist { events.length } == event_count
```

Comment:
I was decided to drop the usage of `insist` for all new specs. We should use the rspec `expect` syntax .